tests/arguments.test-d.ts

```diff
@@ -314,6 +314,12 @@ program
     expectAssignable<OptionValues>(options);
   });
 
+// current Command should be `this` context
+program
+  .action(function (){
+    expectType<Command>(this)
```

Comment:
Thanks for adding a test.

This test looks a bit weak, or subtle. It is accurate for the unconfigured `program`, because the type of `program` is `Command<>` (no options, not arguments) but I had to think about that. For example, adding an option to the program will make the test fail because it won't be plain `<Command>` anymore.

Maybe something more explicit that shows the inferred typings:
```ts
program
  .option('-o, --opt')
  .argument('<arg>')
  .action(function (){
    expectType<Command<[string], { opt?: true}> >(this)
  })
```

Reply:
You are right! I've updated the test